data/wikipron/src/scrape.py

```diff
@@ -71,6 +82,12 @@ def main() -> None:
     }
 
     for iso639_code in languages:
+
+        ###Checks for language restrictions
+        if args.restriction and iso639_code not in args.restriction:
```

Comment:
Here's an alternative that's maybe more complicated but more efficient. Having written it out I'm not sure it's an improvement, so take this with a grain of salt.

    codes = list(languages.keys())
    if args.restriction:
       rset = frozenset(args.restriction)
       lset = frozenset(codes)
       if rset - lset: 
          # fatal error here: log the missing keys and exit(1)
      codes = list(rset & lset)

Then the loop is just:

     for code in codes:
         language_settings = languages[code]  
         ...

Reply:
Definitely better. Throwing it in. (Though isn't searching rather cheap here?)